### ISSUE TITLE
Avoid recycling VLAN ID too quickly

### ISSUE THREAD

**Issue by chlunde** [aws/amazon-vpc-resource-controller-k8s#70]
**What would you like to be enhanced**:
I'm debugging an AWS CNI issue, where we see pods stuck with just vlanXXXX and no vlan.eth.$vlanid@ethN interface:

https://github.com/aws/amazon-vpc-cni-k8s/issues/1644#issuecomment-931109499

I notice that the lowest available VLAN ID is used in this code:

https://github.com/aws/amazon-vpc-resource-controller-k8s/blob/f625e3890791d9d882907bcb8d14afe04d1aac24/pkg/provider/branch/trunk/trunk.go#L583-L595

We suspect this causes the ID to be reused before the cleanup process on the node.

**Why is the change needed and what use case will it solve**:

I think it would be easier to debug if the IDs were allocated sequentially instead. This would decrease the likelyhood of race conditions too, and possibly fix the issue. Regardless of if this is the root cause, I think the change would be nice to do. Is there a downside to this?

**Comment by abhipth** [comment]
@chlunde I don't see any downside to this approach other than this could mask issues as the one you described [here](https://github.com/aws/amazon-vpc-cni-k8s/issues/1644) and a user could still run into the issue with high Pod Churn.

A more concrete short term fix would be more appropriate in [amazon-vpc-cni-k8s](https://github.com/aws/amazon-vpc-cni-k8s) plugin. Also, the vpc-resource-controller release process is more time consuming as compared to `amazon-vpc-cni-k8s` which runs on Data Plane.  

And as discussed in the thread the long term solution would be to have a persistent storage of container ID to VLAN ID.

Let me know your thought!

**Comment by chlunde** [comment]
@abhipth yeah, I can see why rolling out a new release here would be more difficult. Is it normally bound to a specific EKS version or is it fully AWS managed and rolled out independently?

I haven't seen any draft PR/design doc for the other fix, so I left this open as I was not sure if this should be combined with the other fix to avoid delays on rescheduling the same pod on the same node. Or will the VLAN allocation be fully moved to the node?

Either way, feel free to close this issue for now. I'm not sending a PR because I don't this I can run/test this code, I was just eager to get it fixed :smile:
